### PR TITLE
Shorten README.md and add FAQ

### PR DESCRIPTION
**Pull request by geoalgo**
*Description of changes:* this change reduces significantly the size of the README.md by keeping mostly an install section, a small getting-started section and links to extra references. A FAQ have been created which contains the information that was previously in the README.md (how to setup AWS, what output is available etc).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by geoalgo** [comment]
PS: the FAQ section has been written offline with @mseeger.

File: docs/faq.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+   reporter(step=step, metric=float(step) / 3)
+   
+# [tune-metric]: {"step": 0, "metric": 0.0, "st_worker_timestamp": 1644311849.6071281, "st_worker_time": 0.0001048670000045604, "st_worker_iter": 0}
+# [tune-metric]: {"step": 1, "metric": 0.3333333333333333, "st_worker_timestamp": 1644311849.6071832, "st_worker_time": 0.00015910100000837701, "st_worker_iter": 1}
+# [tune-metric]: {"step": 2, "metric": 0.6666666666666666, "st_worker_timestamp": 1644311849.60733, "st_worker_time": 0.00030723599996917983, "st_worker_iter": 2}
+```
+
+### <a name="multiple-gpus"></a> How can I utilize multiple GPUs?
+
+To utilize multiple GPUs you can either use the backend `LocalBackend` which will run on the GPU available in a local machine. You can also run on a remote AWS machine with multiple GPUs using the local backend and the remote launcher, see [I don’t want to wait, how can I launch the tuning on a remote machine?](https://quip-amazon.com/TGFSA6CJ7XD0#temp:C:CUV684fe27b049bbc7bd3d1a9877) or run with the SageMaker backend which spins-up one training job per trial.
```
Comment (mseeger) [inline docs/faq.md:67]: Replace link to Quip with link to other section here

File: docs/faq.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+When using SageMaker backend or tuning remotely, we use [SageMaker checkpoint mechanism](https://docs.aws.amazon.com/sagemaker/latest/dg/model-checkpoints.html) under the hood to sync local checkpoints to s3. Checkpoints are synced to  `s3://{sagemaker-default-bucket}/syne-tune/{tuner-name}/{trial-id}/`, where `sagemaker-default-bucket` is the default bucket for SageMaker (can also be configured).
+
+### <a name="schedulers-checkpointing"></a> Which schedulers make use of checkpointing?
+
+Checkpointing means storing the state of a trial (i.e., model parameters, optimizer or learning rate scheduler parameters), so that it can be paused and potentially resumed at a later point in time, without having to start training from scratch. Syne Tune checkpointing support is detailed in https://github.com/awslabs/syne-tune/blob/main/docs/benchmarks.md#checkpointing and https://github.com/awslabs/syne-tune/blob/main/examples/training_scripts/checkpoint_example/checkpoint_example.py. Once checkpointing to and from a local file is added to a training script, Syne Tune is managing the checkpoints (e.g., copy to/from cloud storage for distributed back-ends).
```
Comment (mseeger) [inline docs/faq.md:93]: Nicer links: [here](https://github.com/awslabs/syne-tune/blob/main/docs/benchmarks.md#checkpointing) and [checkpoint_example.py](https://github.com/awslabs/syne-tune/blob/main/examples/training_scripts/checkpoint_example/checkpoint_example.py)

Comment (mseeger) [reply docs/faq.md:93]: I am a bit surprised these are all absolute links. Don't we want to use relative links, not starting with https://github.com ?

File: docs/faq.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+* Population-based training: `PopulationBasedTraining`
+    PBT does not work without checkpointing.
+* Synchronous Hyperband: `SynchronousGeometricHyperbandScheduler`
+    This code runs without checkpointing, but wastes effort in the same sense as promotion-based asynchronous Hyperband
+
+### <a name="tuner-checkpointing"></a> Is the tuner checkpointed?
+
+Yes. When performing the tuning, the tuner state is regularly saved on the experiment path under `tuner.dill`
+(every 10 seconds which can be configured with `results_update_interval`).
+This allows to use spot-instances when running a tunning remotely with the remote launcher. It also allows to
```
Comment (mseeger) [inline docs/faq.md:108]: I am just wondering. Is this spot instance support fully functional? I've not yet seen an example. It would make a nice extra small tutorial.

File: docs/faq.md [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+### <a name="tuner-checkpointing"></a> Is the tuner checkpointed?
+
+Yes. When performing the tuning, the tuner state is regularly saved on the experiment path under `tuner.dill`
+(every 10 seconds which can be configured with `results_update_interval`).
+This allows to use spot-instances when running a tunning remotely with the remote launcher. It also allows to 
+resume a past experiment or analyse the state of scheduler at any point.
+
+### <a name="trial-output"></a> Where can I find the output of my trials?
+
+When running  `LocalBackend` locally, results of trials are saved under `~/syne-tune/{tuner-name}/{trial-id}/` and contains the following files:
```
Comment (mseeger) [inline docs/faq.md:113]: You should add that if the local backend is run remotely, all this information will be synced to S3, even including the logs and checkpoints.

File: docs/faq.md [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+
+### <a name="trial-execution"></a> How are trials evaluated when evaluating trials on a local machine?
+
+When trials are executed locally (e.g. when `LocalBackend` is used), each trial is evaluated as a different sub-process.
+
+### <a name="tuning-output"></a> What does the the output of the tuning contain?
+
+When running locally, Syne Tune store tuning results on  `~/syne-tune/{tuner-name}/`. This directory will contain `metadata.json`, `results.csv.zip`, and `tuner.dill` which are respectively metadata of the tuning job, results obtained at each time-step and state of the tuner.
+
+When running remotely, those files are saved by default under `s3://{sagemaker-default-bucket}/syne-tune/{tuner-name}/`.
```
[the remark this answers is not in the document]
Comment (mseeger) [reply docs/faq.md:83]: NOTE: This is redundant with "Where can I find the output of the tuning?". Can you merge these?

File: docs/faq.md (hunk 4 above)
Comment (mseeger) [reply docs/faq.md:113]: This is a nice answer here. To me, it feels that it subsumes a few of the others. You here say what is stored for the experiment, and in which subdirectories. This includes results, model checkpoints (optional), and logs, the latter per trial. For remote tuning and local backend, all of this is synced to S3 periodically.

File: docs/faq.md [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+    metadata={"tag": "special-tag", "user": "alice"},
+)
+```
+
+All Syne Tune and user metadata are saved when the tuner starts under `metadata.json`.
+
+
+### <a name="logging-additional-information"></a> How do I append additional information to the results which are stored? 
+
+Results are processed and stored by callbacks passed to `Tuner`, in particular see https://github.com/awslabs/syne-tune/blob/main/syne_tune/tuner_callback.py#L80. In order to add more information to these results, you can inherit from `StoreResultsCallback`. A good example is given in https://github.com/awslabs/syne-tune/blob/main/syne_tune/optimizer/schedulers/searchers/searcher_callback.py#L63.
```
Comment (mseeger) [inline docs/faq.md:190]: Would be nice to have shorter links [short](full). Sorry, maybe I did not do this, but did not know how i Quip.

File: docs/faq.md [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+
+If you run experiments with tabulated benchmarks using the `SimulatorBackend`, as demonstrated in https://github.com/awslabs/syne-tune/blob/main/examples/launch_nasbench201_simulated.py, results are stored by `SimulatorCallback` instead, and you need to inherit from this class, as shown in https://github.com/awslabs/syne-tune/blob/main/syne_tune/optimizer/schedulers/searchers/searcher_callback.py#L88.
+
+### <a name="remote-tuning"></a> I don’t want to wait, how can I launch the tuning on a remote machine?
+
+You can use the remote launcher to launch a tuning on a remote machine. The remote launcher supports both `LocalBackend` and `SagemakerBackend`. In the former case, multiple trials will be evaluated on the remote machine (one use-case being to use a beefy machine), in the latter case trials will be evaluated as separate SageMaker training jobs.
+
+See for an example on how to run tuning with the remote launcher: https://github.com/awslabs/syne-tune/blob/main/examples/launch_height_sagemaker_remotely.py
```
Comment (mseeger) [inline docs/faq.md:198]: Short link

File: docs/faq.md [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+
+### <a name="experiment-parallel"></a> How can I run many experiments in parallel?
+
+You can call the remote launcher multiple times to schedules a list of experiments. In some case, you will want more flexibility and directly write your experiment loop, you can check https://github.com/awslabs/syne-tune/tree/main/benchmarking/benchmark_loop for an example.
```
Comment (mseeger) [inline docs/faq.md:202]: Short link

File: docs/faq.md [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+
+### <a name="results-remote-tuning"></a> How can I access results after tuning remotely?
+
+You can either call `load_experiment("{tuner-name}")` which will download files from s3 if the experiment is not found locally. You can also sync directly files from s3 under `~/syne-tune/` folder in batch for instance by running:
+
+```
+aws s3 sync s3://{sagemaker-default-bucket}/syne-tune/{tuner-name}/ ~/syne-tune/
```
Comment (mseeger) [inline docs/faq.md:209]: This will be tedious when you ran lots of experiments, maybe random repeats etc. If experiment-name is used, I have some instructions in README.txt of ExperimentToolkit how to do a bulk download and also filter out things you do not want (e.g., all the logs and checkpoints, the dill files, etc.).

Happy to add this here.

File: docs/faq.md [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+* Population based training (PBT): https://github.com/awslabs/syne-tune/blob/main/examples/launch_pbt.py
+* Multi-objective asynchronous successive halving (MOASHA): https://github.com/awslabs/syne-tune/blob/main/examples/launch_moasha_instance_tuning.py
+* Constrained Bayesian optimization: https://github.com/awslabs/syne-tune/blob/main/examples/launch_bayesopt_constrained.py
+* Bayesian optimization by density-ratio estimation (BORE): https://github.com/awslabs/syne-tune/blob/main/syne_tune/optimizer/baselines.py#L75
+* Regularized evolution: https://github.com/awslabs/syne-tune/blob/main/syne_tune/optimizer/baselines.py#L87
+* Median stopping rule: https://github.com/awslabs/syne-tune/blob/main/syne_tune/optimizer/schedulers/median_stopping_rule.py
+* Synchronous Hyperband: https://github.com/awslabs/syne-tune/blob/main/syne_tune/optimizer/schedulers/synchronous/hyperband_impl.py#L42
+* Transfer learning schedulers: https://github.com/awslabs/syne-tune/blob/main/examples/launch_nas201_transfer_learning.py
+* Wrappers for Ray Tune schedulers: https://github.com/awslabs/syne-tune/blob/main/examples/launch_height_ray.py
+
```
Comment (mseeger) [inline docs/faq.md:247]: Add a line advertising the short names in baselines.py. Sorry, I forgot to do that back then.

File: README.md [hunk 11]
```diff
@@ -51,267 +51,72 @@ To run all tests whose name begins with `test_async_scheduler`, you can use the
 pytest -k test_async_scheduler
 ```
 
-## How to enable tuning and tuning script conventions
 
-This section describes how to enable tuning an endpoint script. In particular, we describe:
+## Getting started
```
Comment (mseeger) [inline README.md:55]: List above syne-tune[X]: Add 'kde' and 'blackbox-repository' as tags

Comment (mseeger) [reply README.md:55]: Maybe also say that 'raytune' installs all Ray dependencies, which is quite a big chunk.

Comment (mseeger) [reply README.md:55]: Comment on 'pytest -k ...": Is this needed?

**Review by mseeger — COMMENTED** [review]
This looks super clean now. Thanks for driving this.

File: docs/faq.md [hunk 12]
```diff
@@ -0,0 +1,252 @@
...
+Since trials may be paused and resumed (either by schedulers or when using spot-instances), the user has the possibility to checkpoint intermediate results to avoid starting computation from scratch. Model outputs and checkpoints must be written into a specific local path given by the command line argument `st_checkpoint_dir`. Saving/loading model checkpoint from this directory enables to save/load the state when the job is stopped/resumed (setting the folder correctly and uniquely per trial is the responsibility of the backend), see [checkpoint_example.py](https://github.com/awslabs/syne-tune/blob/main/examples/training_scripts/checkpoint_example/checkpoint_example.py) to see a fully working example of a tuning script with checkpoint enabled.
+
+When using SageMaker backend or tuning remotely, we use [SageMaker checkpoint mechanism](https://docs.aws.amazon.com/sagemaker/latest/dg/model-checkpoints.html) under the hood to sync local checkpoints to s3. Checkpoints are synced to  `s3://{sagemaker-default-bucket}/syne-tune/{tuner-name}/{trial-id}/`, where `sagemaker-default-bucket` is the default bucket for SageMaker (can also be configured).
```
[the remark this answers is not in the document]
Comment (geoalgo) [reply docs/faq.md:89]: It depends whether `store_logs_localbackend` is set to true or not.

File: README.md (hunk 11 above)
Comment (geoalgo) [reply README.md:55]: The line for pytest was added by Iaro IIRC, I added kde but blackbox-repo as we dont mention it yet too much in our doc (users can get it by just doing extra).

File: docs/faq.md (hunk 9 above)
Comment (geoalgo) [reply docs/faq.md:209]: I updated with a way to download only the desired files.

File: docs/faq.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (geoalgo) [reply docs/faq.md:67]: Actually, it depends. For the local backend you can also run all workers on all GPUs when you deactivate the GPU rotation.

File: docs/faq.md [hunk 13]
```diff
@@ -187,34 +197,43 @@ All Syne Tune and user metadata are saved when the tuner starts under `metadata.
...
+You can call the remote launcher multiple times to schedules a list of experiments. In some case, you will want more flexibility and directly write your experiment loop, you can check 
+[benchmark_loop](https://github.com/awslabs/syne-tune/tree/main/benchmarking/benchmark_loop) for an example.
 
 ### <a name="results-remote-tuning"></a> How can I access results after tuning remotely?
 
 You can either call `load_experiment("{tuner-name}")` which will download files from s3 if the experiment is not found locally. You can also sync directly files from s3 under `~/syne-tune/` folder in batch for instance by running:
 
 ```
-aws s3 sync s3://{sagemaker-default-bucket}/syne-tune/{tuner-name}/ ~/syne-tune/
+aws s3 sync s3://{sagemaker-default-bucket}/syne-tune/{tuner-name}/ ~/syne-tune/  --include "*"  --exclude "*tuner.dill"
```
Comment (mseeger) [inline docs/faq.md:226]: This is fine. My example was rather doing --exclude "*", then include results.tar.gz and metadata.json. But this is fine as well, as long as the user sees how this is done.

File: docs/faq.md [hunk 14]
```diff
@@ -187,34 +197,43 @@ All Syne Tune and user metadata are saved when the tuner starts under `metadata.
...
 ```
 
+To get all results without the tuner state (you can ommit the include and exclude if you also want to include the tuner state).
```
Comment (mseeger) [inline docs/faq.md:229]: omit

**Review by mseeger — APPROVED** [review]
